File: src/main/java/ru/fullrest/mfr/plugins_configuration_utility/model/repository/PropertiesRepository.java
package ru.fullrest.mfr.plugins_configuration_utility.model.repository;

import org.springframework.data.repository.CrudRepository;
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.Properties;
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.PropertyKey;

/**
 * Created on 02.11.2018
 *
 * @author Alexey Plekhanov
 */
public interface PropertiesRepository extends CrudRepository<Properties, Integer> {

    //    @Query("SELECT p.value FROM Properties p WHERE p.key=?1")
    Properties findByKey(PropertyKey key);
}

File: desktop/application/src/main/kotlin/ru/fullrest/mfr/plugins_configuration_utility/model/repository/PropertiesRepository.kt
package ru.fullrest.mfr.plugins_configuration_utility.model.repository

import org.springframework.data.jpa.repository.JpaRepository
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.Properties
import ru.fullrest.mfr.plugins_configuration_utility.model.entity.PropertyKey

interface PropertiesRepository : JpaRepository<Properties, Int> {
    fun findByKey(key: PropertyKey): Properties?

    //    @Query(
//        """
//            select case when (count (p) > 0) then true else false end from Properties p
//            where p.key = :key
//        """
//    )
    fun existsByKey(key: PropertyKey): Boolean
}